File: app/src/main/java/com/karol/sezonnazdrowie/model/BootReceiver.java
package com.karol.sezonnazdrowie.model;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = "BootReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            SnzAlarmManager.setAlarms(context, null);
        }
    }
}

File: app/src/main/java/com/karol/sezonnazdrowie/model/BootReceiver.kt
package com.karol.sezonnazdrowie.model

import android.app.AlarmManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.karol.sezonnazdrowie.SnzApplication

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val currentDayProvider = SnzAlarmManager.TimeDataProviderImpl()
            val persistentStorage: PersistentStorage = SharedPreferencesPersistentStorage(context)
            val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager

            SnzAlarmManager.setAlarms(
                context,
                currentDayProvider,
                persistentStorage,
                alarmManager,
                (context.applicationContext as SnzApplication).database
            )
        }
    }
}
